### ISSUE TITLE
Composer doesn't warn about abandoned packages

### ISSUE THREAD

**Issue by marijnvdwerf** [composer/composer#6652]
When I run `composer init`, and search for `guzzle`, I get the following output:

```
Search for a package: guzzle             

Found 15 packages matching guzzle

   [0] guzzlehttp/guzzle
   [1] guzzle/guzzle
   [2] ehough/guzzle
   [3] forknetwork/guzzle3
   [4] php-http/guzzle6-adapter
   [5] php-http/guzzle5-adapter
   [6] e-moe/guzzle6-bundle
   [7] trademachines/guzzle5-circuit-breaker
   [8] hrodic/guzzle5-log-subscriber
   [9] kamermans/guzzle4-oauth2-subscriber
  [10] guzzle/parser
  [11] guzzle/common
  [12] guzzle/stream
  [13] guzzle/http
  [14] qbnk/guzzle5-oauth2-subscriber

Enter package # to add, or the complete package name if it is not listed: 
```

And I expected this to happen:
A lot of these packages are marked as abandoned on packagist.com. It would be nice if Composer showed a warning next to those packages.

**Comment by Deamon** [comment]
or maybe do not list them until a specific option is passed to the command ?

**Comment by Jean85** [comment]
The issue is that the current response from Packagist for the search query doesn't hold that information. This is the start of what you would obtain with that search command:

```json
{
	"results": [{
		"name": "guzzlehttp\/guzzle",
		"description": "Guzzle is a PHP HTTP client library",
		"url": "https:\/\/packagist.org\/packages\/guzzlehttp\/guzzle",
		"repository": "https:\/\/github.com\/guzzle\/guzzle",
		"downloads": 42734086,
		"favers": 10120
	}, {
		"name": "guzzle\/guzzle",
		"description": "PHP HTTP client. This library is deprecated in favor of https:\/\/packagist.org\/packages\/guzzlehttp\/guzzle",
		"url": "https:\/\/packagist.org\/packages\/guzzle\/guzzle",
		"repository": "https:\/\/github.com\/guzzle\/guzzle3",
		"downloads": 28036766,
		"favers": 152
	}
]}
```
As you can see, the deprecated field is not shown. Fixing this issue would mean either adding that information on packagist, or either doing many subsequent subcalls for each entry returned from the first search query.

**Comment by Seldaek** [comment]
Should be easy to add the flag to the return value on packagist's end. Then IMO showing a warning next to the name rather than hiding is better.

**Comment by stof** [comment]
maybe also moving them to the end of the list of choices

**Comment by Seldaek** [comment]
Fixed by #6728
